### ISSUE TITLE
Should Molecule equality compare the exact bond order for aromatic bonds?

### ISSUE THREAD

**Issue by andrrizzi** [openforcefield/openforcefield#205]
This snippet fails
```python
file_path = get_data_filename('molecules/MiniDrugBank_tripos.mol2')
molecules = Molecule.from_file(file_path, exception_if_undefined_stereo=False)
molecule = [mol for mol in molecules if mol.name == 'DrugBank_2210'][0]

iupac = molecule.to_iupac()
molecule_copy = Molecule.from_iupac(iupac)
assert molecule.is_isomorphic(molecule_copy)
```
because the molecule has a benzene ring for which the two methods of creating the `Molecule` end up assigning the alternate bond orders 1 and 2 differently.

I'm thinking of changing our condition for an [equal bond](https://github.com/openforcefield/openforcefield/blob/topology/openforcefield/topology/molecule.py#L1872-L1875) from
```python
(x['bond_order'] == y['bond_order']) and
(x['is_aromatic'] == y['is_aromatic']) and
(x['stereochemistry'] == y['stereochemistry'])
```
to
```python
((x['bond_order'] == y['bond_order']) or (x['is_aromatic'] == y['is_aromatic'])) and
(x['stereochemistry'] == y['stereochemistry'])
```

Are there cases in which this wouldn't do what we want?

**Comment by j-wags** [comment]
I think this is a good idea. The only case I can think of where this would fail would be a really exotic molecule, like the one in the second response here: https://www.quora.com/Can-aromatic-compound-have-triple-bonds

Since I seriously doubt we'll run into anything like that, I think your proposed change is great..

**Comment by andrrizzi** [comment]
Fixed in #209 .
